src/bios/flib/src/fs/ext4.rs

```diff
@@ -0,0 +1,671 @@
+        if self.uses_extent_tree() {
+            print_str("Calling extent");
+            return self.get_nth_data_block_extent(block_size, n, partition);
+        } else {
+            print_str("No extent tree");
+            return 0
+        }
+    }
+
+    // Get block number of the nth data block, considering this inode uses an extent tree structure
+    pub fn get_nth_data_block_extent(&self, block_size : u32, n : u32, partition : &Ext4Partition) -> u64 {
+        let mem_offset = self as *const Inode as u32 + 0x28;
+        debug!(mem_offset);
+        return self.explore_next_layer(mem_offset , n , block_size, partition);
+    }
+
+    // Explore next layer, used by get_nth_data_block_extent
+    pub fn explore_next_layer(&self, mem_offset : u32, n : u32, block_size : u32, partition : &Ext4Partition) -> u64 {
+        let header_address = mem_offset as *const Ext4ExtentHeader;
+        let header : &Ext4ExtentHeader;
+        header = unsafe {
+            transmute(header_address)
+        };
+        let leaf_number = header.eh_entries;
+        debug!(leaf_number);
+        debug!(header.eh_depth);
+        // Handle leaves
+        if header.eh_depth == 0 {
+            for i in 0..leaf_number {
+                let extent_addr = (mem_offset + 12*(i+1) as u32) as *const Ext4Extent;
+                debug!(extent_addr as u32);
+                let extent : &Ext4Extent;
+                extent = unsafe {
+                    transmute(extent_addr)
+                };
+                debug!(extent.ee_start_hi);
+                debug!(extent.ee_start_lo);
+                debug!(extent.ee_block);
+                debug!(extent.ee_len);
+                let len = {
+                    if extent.ee_len <= 32768 {
+                        extent.ee_len
+                    } else {
+                        extent.ee_len - 32768
+                    }
+                };
+                if extent.ee_block + len as u32 >= n {
+                    let a = (extent.ee_start_lo as u64 + ((extent.ee_start_hi as u64) << 32)) + (n - extent.ee_block) as u64;
+                    // Return address of n-th block
+                    return a
+                }
+            }
+            return 0
+        } else {
+            // Offset + block_size in memory and iterate over entries recursively
+            for i in 0..leaf_number {
+                let extent_idx : &Ext4ExtentIdx;
+                let extent_idx_addr = (mem_offset + 12*(i+1) as u32) as *const Ext4ExtentIdx;
+                extent_idx = unsafe {
+                    transmute(extent_idx_addr)
+                };
+                let next_block_address = extent_idx.ei_leaf_lo as u64 + (extent_idx.ei_leaf_hi as u64) << 2;
+                // let a = AddressPacket::new((block_size / 512) as u16, mem_offset + block_size, (partition_offset + next_block_address / 512) as u64);
+                let _ = partition.read(next_block_address as u32, block_size, mem_offset + block_size);
```

Comment:
Si ça peut échouer, passe le type de retour en `Result<u64, ()>` en suivant la suggestion: 
```suggestion
                partition.read(next_block_address as u32, block_size, mem_offset + block_size)?;
```

Reply:
Je fais pas te cacher que j'ai pas trop travaillé le fait de handle les erreurs et tout, faudrait probablement que je le fasse